lib/services/android-project-service.ts

```diff
@@ -149,12 +149,23 @@ export class AndroidProjectService extends projectServiceBaseLib.PlatformProject
 				let projectPackageJson: any = this.$fs.readJson(this.$projectData.projectFilePath).wait();
 
 				_.each(AndroidProjectService.REQUIRED_DEV_DEPENDENCIES, (dependency: any) => {
-					let dependencyVersionInProject = projectPackageJson.dependencies[dependency.name] || projectPackageJson.devDependencies[dependency.name];
+					let dependencyVersionInProject = (projectPackageJson.dependencies && projectPackageJson.dependencies[dependency.name]) ||
+						(projectPackageJson.devDependencies && projectPackageJson.devDependencies[dependency.name]);
 
 					if (!dependencyVersionInProject) {
 						this.$npm.install(`${dependency.name}@${dependency.version}`, this.$projectData.projectDir, npmConfig).wait();
-					} else if (!semver.satisfies(dependencyVersionInProject, dependency.version)) {
-						this.$errors.failWithoutHelp(`Your project have installed ${dependency.name} version ${dependencyVersionInProject} but Android platform requires version ${dependency.version}.`);
+					} else {
+						let cleanedVerson = semver.clean(dependencyVersionInProject);
+
+						// The plugin version is not valid. Check node_modules for the valid version.
+						if (!cleanedVerson) {
+							dependencyVersionInProject = this.$fs.exists(path.join(this.$projectData.projectDir, "node_modules", dependency.name, "package.json")).wait() &&
+								this.$fs.readJson(path.join(this.$projectData.projectDir, "node_modules", dependency.name, "package.json")).wait().version;
```

Comment:
Extract this to variable: `path.join(this.$projectData.projectDir, "node_modules", dependency.name, "package.json")`